File: C1/PDOO/Material Complementario/Tema7/ejemplosTema7/ImplementacionEjemplosTema7/src/ejemplodsecuencia/OrganizadoraConciertosBeneficos.java

package ejemplodsecuencia;

//solo se implementa aquello directamente relacionado con el diagrama de secuencia del ejemplo

public class OrganizadoraConciertosBeneficos {
    
    public String comprarEntradas(String nombreCliente, String nombreConcierto, Date fecha, int numeroEntradas){
        
        //envío de mensaje 1 del diagrama. Es necesario declarar c como variable de la clase que devuelve el método
        ConciertoBenefico c= buscarConcierto(nombreConcierto);
        
        //envío de mensaje 2 del diagrama. Es necesario declarar mensaje como variable de la clase que devuelve el método
        String mensaje= c.comprarEntradas(nombreCliente, fecha, numeroEntradas);
        
        return mensaje; 
        
        //ver implementación de envío de mensajes 2.1 a 2.4 en ConciertoBenéfico, método comprarEntradas
    }
    
    public ConciertoBenefico buscarConcierto(String nombreConcierto){
        
    }
    
}

File: C1/PDOO/Material Complementario/Tema7/ejemplosTema7/ImplementacionEjemplosTema7Ruby/lib/ejemplodsecuencia/organizadora_conciertos_beneficos.rb
class OrganizadoraConciertosBeneficos
  def initialize
    
  end
def comprar_entradas(nombre_cliente, nombre_concierto, fecha, numero_entradas)
        
       #envío de mensaje 1 del diagrama. 
       c= buscarConcierto(nombre_concierto)
        
       #envío de mensaje 2 del diagrama. 
       mensaje= c.comprarEntradas(nombre_cliente, fecha, numero_entradas)
        
       mensaje 
        
       #ver implementación de envío de mensajes 2.1 a 2.4 en ConciertoBenéfico, método comprar_entradas
end

def buscarConcierto(nombre_concierto)
  
end
    

end
